src/Microsoft.DotNet.Cli.Utils/NativeMethods.cs

```diff
@@ -0,0 +1,80 @@
+using System;
+using System.Runtime.InteropServices;
+using Microsoft.Win32.SafeHandles;
+
+namespace Microsoft.DotNet.Cli.Utils
+{
+    internal static class NativeMethods
+    {
+        internal static class Windows
+        {
+            internal enum JobObjectInfoClass : uint
+            {
+                JobObjectExtendedLimitInformation = 9,
+            }
+
+            [Flags]
+            internal enum JobObjectLimitFlags : uint
+            {
+                JobObjectLimitKillOnJobClose = 0x2000,
+            }
+
+            [StructLayout(LayoutKind.Sequential)]
+            internal struct JobObjectBasicLimitInformation
+            {
+                public Int64 PerProcessUserTimeLimit;
+                public Int64 PerJobUserTimeLimit;
+                public JobObjectLimitFlags LimitFlags;
+                public UIntPtr MinimumWorkingSetSize;
+                public UIntPtr MaximumWorkingSetSize;
+                public UInt32 ActiveProcessLimit;
+                public UIntPtr Affinity;
+                public UInt32 PriorityClass;
+                public UInt32 SchedulingClass;
+            }
+
+            [StructLayout(LayoutKind.Sequential)]
+            internal struct IoCounters
+            {
+                public UInt64 ReadOperationCount;
+                public UInt64 WriteOperationCount;
+                public UInt64 OtherOperationCount;
+                public UInt64 ReadTransferCount;
+                public UInt64 WriteTransferCount;
+                public UInt64 OtherTransferCount;
+            }
+
+            [StructLayout(LayoutKind.Sequential)]
+            internal struct JobObjectExtendedLimitInformation
+            {
+                public JobObjectBasicLimitInformation BasicLimitInformation;
+                public IoCounters IoInfo;
+                public UIntPtr ProcessMemoryLimit;
+                public UIntPtr JobMemoryLimit;
+                public UIntPtr PeakProcessMemoryUsed;
+                public UIntPtr PeakJobMemoryUsed;
+            }
+
+            [DllImport("kernel32.dll", CharSet = CharSet.Unicode, SetLastError = true)]
+            internal static extern SafeWaitHandle CreateJobObjectW(IntPtr lpJobAttributes, string lpName);
+
+            [DllImport("kernel32.dll", SetLastError = true)]
+            internal static extern bool SetInformationJobObject(IntPtr hJob, JobObjectInfoClass jobObjectInformationClass, IntPtr lpJobObjectInformation, UInt32 cbJobObjectInformationLength);
+
+            [DllImport("kernel32.dll", SetLastError = true)]
+            internal static extern bool AssignProcessToJobObject(IntPtr hJob, IntPtr hProcess);
+        }
+
+        internal static class Posix
+        {
+            [DllImport("libc", SetLastError = true)]
+            internal static extern int getpgid(int pid);
```

Comment:
`getpgid` is always successful. SetLastError isn't needed.

Reply:
The POSIX specification says it can fail (e.g. `ESRCH`), although the only place we're calling it is for the current pid which should always be successful.  I believe, however, that the P/Invoke signature should always be correct for the API itself and not for how it is invoked in case someone uses it differently in the future and expects correctness.

Reply:
Ah yes!